reputation/reputation.go

```diff
@@ -0,0 +1,576 @@
+
+	rows, err := db.Query(queryString)
+	if err != nil {
+		log.Printf("%q: %s\n", err, queryString)
+		return bestRep, SelectError.Wrap(err)
+	}
+	defer rows.Close()
+
+	transformedRows, err := iterOnDBRows(rows)
+	if err != nil {
+		log.Printf("%q: %s\n", err, queryString)
+		return bestRep, SelectError.Wrap(err)
+	}
+
+	for _, row := range transformedRows {
+		bestRep = bestRep.endian(row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Printf("%q: %s\n", err, queryString)
+		return bestRep, SelectError.Wrap(err)
+	}
+
+	return bestRep, nil
+}
+
+type mutOp string
+
+// coproduct/sum type for the mutation operation type
+const (
+	increment mutOp = "increment"
+	decrement mutOp = "decrement"
+	overWrite mutOp = "overWrite"
+)
+
+// performOp performs an operation that is passed to it on a value with the scalar
+func performOp(op mutOp, value int, scalar int) int {
+	switch op {
+	case increment:
+		value = value + scalar
+	case decrement:
+		value = value - scalar
+	case overWrite:
+		value = scalar
+	}
+
+	return value
+}
+
+type column string
+
+// coproduct/sum type for the column type
+const (
+	uptimeColumn             column = "uptime"
+	auditSuccessColumn       column = "auditSuccess"
+	auditFailColumn          column = "auditFail"
+	latencyColumn            column = "latency"
+	amountOfDataStoredColumn column = "amountOfDataStored"
+	falseClaimsColumn        column = "falseClaims"
+	shardsModifiedColumn     column = "shardsModified"
+)
+
+// morphism is the name of this function becuse it does not directly change the NodeReputationRecord (more map/functor like)
+func (row NodeReputationRecord) morphism(col column, op mutOp, scalar int) NodeReputationRecord {
+	switch col {
+	case uptimeColumn:
+		row.uptime = performOp(op, row.uptime, scalar)
+	case auditSuccessColumn:
+		row.auditSuccess = performOp(op, row.auditSuccess, scalar)
+	case auditFailColumn:
+		row.auditFail = performOp(op, row.auditFail, scalar)
+	case latencyColumn:
+		row.latency = performOp(op, row.latency, scalar)
+	case amountOfDataStoredColumn:
+		row.amountOfDataStored = performOp(op, row.amountOfDataStored, scalar)
+	case falseClaimsColumn:
+		row.falseClaims = performOp(op, row.falseClaims, scalar)
+	case shardsModifiedColumn:
+		row.shardsModified = performOp(op, row.shardsModified, scalar)
+	}
+
+	return row
+}
+
+// NodeReputationRecordMorphism this is more like fmap because the slice is the functor, returns a new NodeReputationRecord slice
+func NodeReputationRecordMorphism(rows []NodeReputationRecord, col column, op mutOp, scalar int) []NodeReputationRecord {
+	var res []NodeReputationRecord
+
+	for _, row := range rows {
+		res = append(res, row.morphism(col, op, scalar))
+	}
+
+	return res
+}
+
+// NewReputationRow this is the apply function for the reputation row struct, returns a new NodeReputationRecord
+func NewReputationRow(name string) NodeReputationRecord {
+	return NodeReputationRecord{name, "", 0, 0, 0, 0, 0, 0, 0}
+}
```

Comment:
i don't remember where dennis said this but i definitely agree it's worth breaking all this logic up into multiple files